### PR TITLE
Record memory flow

### PR DESCRIPTION
**Pull request by JamesTwisleton**

### CODE REVIEW

File: components/ChooseSong.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react'
+import { Container, Row, Col, Table, Form } from 'react-bootstrap/'
+import PropTypes from 'prop-types'
+import { searchSpotifyTracks } from '../utils/fetcher'
+import useAuth from '../utils/useAuth'
+const AddMemory = () => {
```
Comment (ChangoFoster) [inline components/ChooseSong.js:6]: Would chuck in a space between the imports and everything else. Plus I'd change AddMemory to ChooseSong

File: components/ChooseSong.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        } catch (error) {
+          console.log(error)
+        }
+      }
+    }
+    if (token) {
+      searchSpotify()
+    }
+  }, [song, token])
+  return (<>
```
Comment (ChangoFoster) [inline components/ChooseSong.js:26]: Add a space between useEffect and the return

Comment (ChangoFoster) [reply components/ChooseSong.js:26]: Plus you don't need the fragment if it's all wrapped in one html tag (the Container)

File: components/RecordMemory.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import useAuth from '../utils/useAuth'
+// import PropTypes from 'prop-types'
+import { Row, Col, Container, Button } from 'react-bootstrap/'
+import { useRouter } from 'next/router'
+
+const RecordMemory = () => {
+  useAuth()
```
Comment (ChangoFoster) [inline components/RecordMemory.js:8]: I think you'll still need something like this to make the auth work as expected:

const [, token] = useAuth()

if (!token) {
return <p>Loading</p>
}

File: components/RecordMemory.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const router = useRouter()
+  return (<>
```
Comment (ChangoFoster) [inline components/RecordMemory.js:10]: Don't think you need the fragment here

**Review by ChangoFoster — COMMENTED** [review]
A few styling things and the useAuth changes but I think this would all work anyway. BTW this is sort of what I used to make the hook. It's a bit fancier than the one I made as it uses a redux style provider: https://usehooks.com/useAuth/

File: components/ChooseSong.js (hunk 1 above)
Comment (JamesTwisleton) [reply components/ChooseSong.js:6]: Done!

File: components/ChooseSong.js (hunk 2 above)
Comment (JamesTwisleton) [reply components/ChooseSong.js]: Nice!

File: components/RecordMemory.js (hunk 3 above)
Comment (JamesTwisleton) [reply components/RecordMemory.js]: I _think_ I've got that now

File: components/RecordMemory.js (hunk 4 above)
Comment (JamesTwisleton) [reply components/RecordMemory.js]: Indeed!

File: components/RecordMemory.js (hunk 3 above)
Comment (ChangoFoster) [reply components/RecordMemory.js]: Yep, all looks good

**Comment by ChangoFoster** [comment]
Reckon it's all good to merge whenever

**Pull request merged**